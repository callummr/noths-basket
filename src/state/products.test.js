import reducer, { setProducts } from './products'

describe('products reducer', () => {
  it('should match default snapshot', () => {
    expect(reducer()).toMatchSnapshot()
  })

  it('should match setProducts snapshot', () => {
    const setProductsAction = setProducts([
      { id: '004', name: 'Test product', price: 10.5 },
    ])

    expect(reducer({}, setProductsAction)).toMatchSnapshot()
  })
})

describe('products actions', () => {
  describe('setProducts', () => {
    it('should match snapshot', () => {
      expect(
        setProducts([{ id: '004', name: 'Test product', price: 10.5 }])
      ).toMatchSnapshot()
    })
  })
})
