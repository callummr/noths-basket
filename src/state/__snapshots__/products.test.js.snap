// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`products actions setProducts should match snapshot 1`] = `
Object {
  "payload": Array [
    Object {
      "id": "004",
      "name": "Test product",
      "price": 10.5,
    },
  ],
  "type": "noths/products/SET_PRODUCTS",
}
`;

exports[`products reducer should match default snapshot 1`] = `
Array [
  Object {
    "id": "001",
    "name": "Travel Card Holder",
    "price": 9.25,
  },
  Object {
    "id": "002",
    "imageUrl": "/img/002.jpg",
    "name": "Personalised Cufflinks",
    "price": 45,
  },
  Object {
    "id": "003",
    "imageUrl": "/img/003.jpg",
    "name": "Kids T-shirt",
    "price": 19.95,
  },
]
`;

exports[`products reducer should match setProducts snapshot 1`] = `
Array [
  Object {
    "id": "004",
    "name": "Test product",
    "price": 10.5,
  },
]
`;
